crates/evm/src/executor/fork/cache.rs

```diff
@@ -267,22 +273,13 @@ impl MemDb {
                 } else if acc.info.code_hash.is_zero() {
                     acc.info.code_hash = KECCAK_EMPTY;
                 }
-                accounts.insert(add, acc.info);
+                accounts.insert(add, acc.info.clone());
 
                 let acc_storage = storage.entry(add).or_default();
-                if acc.status.contains(AccountStatus::Created) {
+                if is_newly_created {
                     acc_storage.clear();
                 }
-                for (index, value) in acc.storage {
-                    if value.present_value() == U256::from(0) {
-                        acc_storage.remove(&index);
-                    } else {
-                        acc_storage.insert(index, value.present_value());
-                    }
-                }
-                if acc_storage.is_empty() {
-                    storage.remove(&add);
-                }
+                acc_storage.extend(acc.storage.into_iter().map(|(k, v)| (k, v.present_value())));
```

Comment:
I would leave the previous code here, as it does some cleanup of current storage. It probably didn't change any behaviour but it uses a little bit less memory.